actors/builtin/power/power_state.go

```diff
@@ -146,10 +146,19 @@ func (st *State) AddToClaim(s adt.Store, miner addr.Address, power abi.StoragePo
 		return errors.Errorf("no claim for actor %v", miner)
 	}
 
-	st.TotalNetworkPower = big.Add(st.TotalNetworkPower, power)
+	prevBelow := claim.Power.LessThan(ConsensusMinerMinPower)
 
 	claim.Power = big.Add(claim.Power, power)
 	claim.Pledge = big.Add(claim.Pledge, pledge)
+
+	stillBelow := claim.Power.LessThan(ConsensusMinerMinPower)
+
+	if !prevBelow && !stillBelow {
+		st.TotalNetworkPower = big.Add(st.TotalNetworkPower, power)
+	} else if prevBelow && !stillBelow {
+		st.TotalNetworkPower = big.Add(st.TotalNetworkPower, claim.Power)
+	}
```

Comment:
This doesn't appear to handle when a claim is reduced (`power` is negative). That's handled here rather than a duplicative `subtractFromClaim` method. I think you want:
```
if prevBelow != stillBelow { add power }
```

Reply:
argh good catch alex